elfo-network/src/socket/raw/tcp/mod.rs

```diff
@@ -6,7 +6,76 @@ use futures::Stream;
 use tokio::net::{TcpListener, TcpStream};
 use tracing::warn;
 
-pub(super) use tokio::net::tcp::{OwnedReadHalf, OwnedWriteHalf};
+pub(super) use tokio::net::tcp::OwnedReadHalf;
+
+use std::{
+    io::{IoSlice, Result as IoResult},
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tokio::io::AsyncWrite;
+
+#[cfg(target_os = "linux")]
+mod metrics;
+
+#[cfg(target_os = "linux")]
+use self::metrics::TcpMetrics;
+
+pub(in crate::socket) struct OwnedWriteHalf {
+    inner: tokio::net::tcp::OwnedWriteHalf,
+    #[cfg(target_os = "linux")]
+    previous_metrics: TcpMetrics,
```

Comment:
Simply `metrics`. It becomes "previous" only on `collect_transport_specific_metrics()` call and even there it immediately updates to the actual ones